etna/whatson/models.py

```diff
@@ -685,6 +685,34 @@ def primary_access_type(self):
         if primary_access := self.event_access_types.first():
             return primary_access.access_type
 
+    @cached_property
+    def date_time_range(self):
+        delta = self.end_date - self.start_date
+        # One session on one date where start and end times are the same
+        # return eg. Monday 1 January 2024, 19:00
+        if (self.start_date == self.end_date) and (len(self.sessions.all()) == 1):
+            return self.start_date.strftime("%A %-d %B %Y, %H:%M")
+        # One session on one date where there are values for both start time and end time
+        # eg. Monday 1 January 2024, 19:00–20:00 (note this uses an en dash)
+        if (delta.days == 0 and delta.seconds > 0) and (len(self.sessions.all()) == 1):
+            return (
+                self.start_date.strftime("%A %-d %B %Y, %H:%M")
+                + "–"
+                + self.end_date.strftime("%H:%M")
+            )
+        # Multiple sessions on one date
+        # Eg. Monday 1 January 2024
+        if (delta.days == 0) and (len(self.sessions.all()) > 1):
+            return self.start_date.strftime("%A %-d %B %Y")
+        # Event has multiple dates
+        # Eg. 1 January 2024 to 5 January 2024
+        if delta.days > 0:
+            return (
+                self.start_date.strftime("%-d %B %Y")
+                + " to "
+                + self.end_date.strftime("%-d %B %Y")
+            )
```

Comment:
Suggestion: assign the format strings to variables so that they can be reused and are self-documenting. This should make this method a little easier to read at a glance.

It might be nice to use f-strings also, e.g.:

```python
f"{self.start_date.strftime(format_no_time)} to {self.end_date.strftime(format_no_time)}"
```

Reply:
variables added and f-strings added in https://github.com/nationalarchives/ds-wagtail/pull/1470/commits/1238a0ca1875a7be135bf6da42d4f00d7d83375b